blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/astyanax/AstyanaxWriter.java

```diff
@@ -150,10 +169,37 @@ public void insertFull(Collection<? extends IMetric> metrics) throws ConnectionE
         }
     }
 
+    private void insertDelayedLocator(IMetric metric, MutationBatch mutationBatch, boolean isTrackingDelayedMetrics, Clock clock) {
+        Locator locator = metric.getLocator();
+
+        if (isTrackingDelayedMetrics) {
+
+            long delay = clock.now().getMillis() - metric.getCollectionTime();
+            if (delay > MAX_AGE_ALLOWED) {
+
+                //track locator for configured granularity level. to re-roll only the delayed locator's for that slot
+                int slot = DELAYED_METRICS_STORAGE_GRANULARITY.slot(metric.getCollectionTime());
+                if (!AstyanaxWriter.isDelayedLocatorForASlotCurrent(slot, locator)) {
+
+                    insertDelayedLocator(DELAYED_METRICS_STORAGE_GRANULARITY, slot, locator, mutationBatch);
+                    AstyanaxWriter.setDelayedLocatorForASlotCurrent(slot, locator);
+                }
+            }
+        }
+    }
+
+
     // numeric only!
     public final void insertLocator(Locator locator, MutationBatch mutationBatch) {
-                mutationBatch.withRow(CassandraModel.CF_METRICS_LOCATOR, (long) Util.getShard(locator.toString()))
-                        .putEmptyColumn(locator, TenantTtlProvider.LOCATOR_TTL);
+        mutationBatch.withRow(CassandraModel.CF_METRICS_LOCATOR, (long) Util.getShard(locator.toString()))
+                .putEmptyColumn(locator, TenantTtlProvider.LOCATOR_TTL);
+    }
+
+    // numeric only!
+    public final void insertDelayedLocator(Granularity g, int slot, Locator locator, MutationBatch mutationBatch) {
```

Comment:
You have two methods with the name "insertDelayedLocator", one public, one private, but one behaves quite differently than the other. Can we clarify the difference in the method name?

And why is it that the public one does not check if the delayed locator is current or not?